### PR TITLE
Moving `pallet-asset-tx-payment` from cumulus to substrate

### PR DESCRIPTION
**Pull request by georgesdib**
This fixes issue #688.

The corresponding substrate PR is paritytech/substrate#10127.

@apopiak @NukeManDan

### CODE REVIEW

File: polkadot-parachains/parachains-common/Cargo.toml
Comment (KiChjang) [inline polkadot-parachains/parachains-common/Cargo.toml:23]: Point the branch to [your PR branch here](https://github.com/paritytech/substrate/pull/10127) so that our CI can pick up the correct branch to test your substrate PR against.

Comment (georgesdib) [reply polkadot-parachains/parachains-common/Cargo.toml:23]: The only way I managed to make it work was via patches. This made this repo pass the checks, but it's still failing the substrate repo check, would you have an idea?
Also, technically speaking, this PR does not have to be a companion for the substrate one, because you could merge the substrate PR and that would not break cumulus, it would just mean the pallet is defined again in cumulus.

File: Cargo.toml
Comment (apopiak) [inline Cargo.toml:234]: based on 
```
error: failed to parse manifest at `/builds/parity/substrate/cumulus/Cargo.toml`
Caused by:
  dependency (beefy-gadget) specification is ambiguous. Only one of `git` or `path` is allowed.
```
from here https://gitlab.parity.io/parity/substrate/-/jobs/1214029
Your patching is not quite correct.

Comment (apopiak) [reply Cargo.toml:234]: mmh, should be correct, though
@joao-paulo-parity Any ideas what might be going wrong with the patching?

File: polkadot-parachains/parachains-common/Cargo.toml
Comment (apopiak) [reply polkadot-parachains/parachains-common/Cargo.toml:23]: patching is fine/intended for this
with this PR green I guess I'd be fine just marking it in Substrate as related and get the CI green and the PR merged

File: Cargo.toml
Comment (georgesdib) [reply Cargo.toml:234]: Is it not because it double patches? As in, I have a patch, and then the bot would also run diener and tries to patch again and has a conflict? But this is me guessing.

Comment (apopiak) [reply Cargo.toml:234]: That makes sense.

**Comment by georgesdib** [comment]
After the substrate merge, this should now be in a good shape for a merge too. Please let me know if you disagree/suggestions?

**Review by apopiak — APPROVED** [review]
LGTM, would wait for someone else to approve as well, though.

**Review by bkchr — APPROVED** [review]

**Comment by bkchr** [comment]
LGTM.

**Comment by apopiak** [comment]
@georgesdib Thank you again for your contribution! :+1:

**Pull request merged**
